### ISSUE TITLE
ProjectModelServer: Exception when opening project and there no project.lock.json yet (no any snapshot sent)

### ISSUE THREAD

**Issue by abpiskunov** [dotnet/cli#858]
When open existing project that has no project.lock.json file, DTH console shows exception and no any snapshot is sent after that:

```
info: OpenChannel[0]
      Process ID 28428
info: OpenChannel[0]
      Listening on port 2907
info: OpenChannel[0]
      Client accepted 127.0.0.1:2907
info: Microsoft.DotNet.ProjectModel.Server.ProcessingQueue[0]
      Start
info: Microsoft.DotNet.ProjectModel.Server.ProcessingQueue[0]
      OnReceive((1fd25aacbceb4c04bbdf03cfe77b338c, ProtocolVersion, 0) -> {
        "Version": 4
      })
info: Microsoft.DotNet.ProjectModel.Server.ProtocolManager[0]
      Initializing the protocol negotiation.
info: Microsoft.DotNet.ProjectModel.Server.ProtocolManager[0]
      Protocol negotiation successed. Use protocol 4
info: Microsoft.DotNet.ProjectModel.Server.ProtocolManager[0]
      Respond to protocol negotiation.
info: Microsoft.DotNet.ProjectModel.Server.ProcessingQueue[0]
      Send ((1fd25aacbceb4c04bbdf03cfe77b338c, ProtocolVersion, 0) -> {
        "Version": 4
      })
info: Microsoft.DotNet.ProjectModel.Server.ProcessingQueue[0]
      OnReceive((1fd25aacbceb4c04bbdf03cfe77b338c, Initialize, 2) -> {
        "Configuration": "Debug",
        "ProjectFolder": "D:\\temp\\TestProjects\\ConsoleApp1\\src\\ConsoleApp1",
        "Version": 4
      })
info: Microsoft.DotNet.ProjectModel.Server.ProjectManager[0]
      Begin draining inbox.
info: Microsoft.DotNet.ProjectModel.Server.ProjectManager[0]
      Received Initialize
info: Microsoft.DotNet.ProjectModel.Server.ProjectManager[0]
      Set context protocol version to 4
info: Microsoft.DotNet.ProjectModel.Server.ProjectManager[0]
      Finish draining inbox.
fail: Microsoft.DotNet.ProjectModel.Server.ProjectManager[0]
      Error occurred: System.IO.FileNotFoundException: Could not find file 'D:\temp\TestProjects\ConsoleApp1\src\ConsoleApp1\project.lock.json'.
      File name: 'D:\temp\TestProjects\ConsoleApp1\src\ConsoleApp1\project.lock.json'
         at System.IO.Win32FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs)
         at System.IO.Win32FileSystem.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, FileStream parent)
         at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
         at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
         at Microsoft.DotNet.ProjectModel.Graph.LockFileReader.Read(String filePath) in C:\code\cli\src\Microsoft.DotNet.ProjectModel\Graph\LockFileReader.cs:line 20
         at Microsoft.DotNet.ProjectModel.WorkspaceContext.AddLockFileEntry(String projectDirectory, FileModelEntry`1 currentEntry) in C:\code\cli\src\Microsoft.DotNet.ProjectModel\WorkspaceContext.cs:line 205
         at Microsoft.DotNet.ProjectModel.WorkspaceContext.<GetLockFile>b__15_0(String key) in C:\code\cli\src\Microsoft.DotNet.ProjectModel\WorkspaceContext.cs:line 155
         at System.Collections.Concurrent.ConcurrentDictionary`2.AddOrUpdate(TKey key, Func`2 addValueFactory, Func`3 updateValueFactory)
         at Microsoft.DotNet.ProjectModel.WorkspaceContext.GetLockFile(String projectDirectory) in C:\code\cli\src\Microsoft.DotNet.ProjectModel\WorkspaceContext.cs:line 153
         at Microsoft.DotNet.ProjectModel.WorkspaceContext.<AddProjectContextEntry>b__18_1(String path) in C:\code\cli\src\Microsoft.DotNet.ProjectModel\WorkspaceContext.cs:line 238
         at Microsoft.DotNet.ProjectModel.ProjectContextBuilder.Build() in C:\code\cli\src\Microsoft.DotNet.ProjectModel\ProjectContextBuilder.cs:line 162
         at Microsoft.DotNet.ProjectModel.WorkspaceContext.AddProjectContextEntry(String projectDirectory, ProjectContextCollection currentEntry) in C:\code\cli\src\Microsoft.DotNet.ProjectModel\WorkspaceContext.cs:line 242
         at Microsoft.DotNet.ProjectModel.WorkspaceContext.<GetProjectContextCollection>b__13_0(String key) in C:\code\cli\src\Microsoft.DotNet.ProjectModel\WorkspaceContext.cs:line 139
         at System.Collections.Concurrent.ConcurrentDictionary`2.AddOrUpdate(TKey key, Func`2 addValueFactory, Func`3 updateValueFactory)
         at Microsoft.DotNet.ProjectModel.WorkspaceContext.GetProjectContextCollection(String projectPath) in C:\code\cli\src\Microsoft.DotNet.ProjectModel\WorkspaceContext.cs:line 137
         at Microsoft.DotNet.ProjectModel.Server.ProjectSnapshot.Create(String projectDirectory, String configuration, WorkspaceContext workspaceContext, IReadOnlyList`1 projectSearchPaths) in C:\code\cli\src\Microsoft.DotNet.ProjectModel.Server\InternalModels\ProjectSnapshot.cs:line 24
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.UpdateProject() in C:\code\cli\src\Microsoft.DotNet.ProjectModel.Server\ProjectManager.cs:line 250
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.DoProcessLoop() in C:\code\cli\src\Microsoft.DotNet.ProjectModel.Server\ProjectManager.cs:line 159
         at Microsoft.DotNet.ProjectModel.Server.ProjectManager.ProcessLoop() in C:\code\cli\src\Microsoft.DotNet.ProjectModel.Server\ProjectManager.cs:line 121
info: Microsoft.DotNet.ProjectModel.Server.ProcessingQueue[0]
      Send ((1fd25aacbceb4c04bbdf03cfe77b338c, Error, 2) -> {
        "Message": "Could not find file 'D:\\temp\\TestProjects\\ConsoleApp1\\src\\ConsoleApp1\\project.lock.json'.",
        "Path": null,
        "Line": 0,
        "Column": 0
      })
```

**Comment by davidfowl** [comment]
What did it do before?

**Comment by troydai** [comment]
The exception should be suppress first. There should be error in `DependencyDiagnostics` suggesting IDE to restore.

**Comment by abpiskunov** [comment]
continue with normal initialization and initial snapshots, then tooling would know that restore needed based on them . Now we don't have any data.

**Comment by troydai** [comment]
PR is out.

**Comment by troydai** [comment]
Fixed in 68c0edf
